lib/plugins/features/movies/controller.js

```diff
@@ -1,9 +1,56 @@
 'use strict';
 
-const Movie = require('../../../models/movie');
+// Validators
+const Movie     = require('../../../models/movie');
+const MovieList = require('../../../validators/movielist');
+
+// Libraries for SQL requests
+const Boom = require('@hapi/boom');
+const Knex = require('../../../libraries/knex');
 
 exports.create = async (payload) => {
   const movie = await new Movie().save(payload);
 
   return new Movie({ id: movie.id }).fetch();
 };
+
+exports.retrieve = async (query) => {
+  // Perform validation inside method since Hapi doesn't support
+  // validation on GET requests
+  const validation_result = MovieList.validate(query);
+  /* istanbul ignore if */
+  if (validation_result.error) {
+    throw Boom.badRequest(validation_result.error.message, {
+      errorCode: 422  // Joi validation error code
+    });
+  }
+
+  // Use parameters to construct SQL query, store in result
+  const result = Knex.select().from('movies');
```

Comment:
What led you to use Knex directly, rather than making the queries via Bookshelf? There's nothing wrong with that choice, I'm just curious why you made it. (Although I would say that in this situation it's a little confusing, since you're not really using the model.)

Reply:
Actually, one other thought about this: the way you have this written, `query()` is kind of doing two things: one is validating the GET request and transforming the result of the db query into something that can be returned to the client, the other is performing the actual database lookups themselves. It's not a big deal in this situation because everything is pretty short and straightforward, but it's the kind of thing that as the functionality gets more complicated I might think about separating into multiple functions.

Reply:
I changed the implementation to use Bookshelf instead of Knex to make the queries!